pkg/odo/cli/logs/logs.go

```diff
@@ -79,14 +90,42 @@ func (o *LogsOptions) Complete(cmdline cmdline.Cmdline, args []string) error {
 }
 
 func (o *LogsOptions) Validate() error {
+	if o.devMode && o.deployMode {
+		return errors.New("pass only one of --dev or --deploy flags; pass no flag to see logs for both modes")
+	}
 	return nil
 }
 
 func (o *LogsOptions) Run(ctx context.Context) error {
-	containersLogs, err := o.clientset.LogsClient.DevModeLogs(o.componentName, o.Context.GetProject())
+	var logMode logsMode
+	var err error
+	var containersLogs []map[string]io.ReadCloser
+
+	if o.devMode {
+		logMode = DevMode
+	} else if o.deployMode {
+		logMode = DeployMode
+	}
+
+	switch logMode {
+	case DevMode:
+		containersLogs, err = o.clientset.LogsClient.DevModeLogs(o.componentName, o.Context.GetProject())
+	case DeployMode:
+		containersLogs, err = o.clientset.LogsClient.DeployModeLogs(o.componentName, o.Context.GetProject())
+	default:
+		containersLogs, err = o.clientset.LogsClient.AllModeLogs(o.componentName, o.Context.GetProject())
+	}
```

Comment:
Not liking to use the `if..else` **AND** `switch..case`. But I felt a use case for using `default` to show logs for both modes. Open to ideas/suggestions.